src/mongocxx/collection.hpp

```diff
@@ -93,6 +91,16 @@ class MONGOCXX_API collection {
     collection& operator=(collection&&) noexcept;
 
     ///
+    /// Copy constructs a collection.
+    ///
+    collection(const collection&) noexcept;
```

Comment:
I don't think the copy constructor and copy assignment operator should be noexcept. They allocate memory via make_unique, so could possibly throw.